modules/packages/Python.chpl

```diff
@@ -2418,13 +2545,18 @@ module Python {
       :arg item: The item to set.
     */
     proc set(idx: int, item: eltType) throws {
-      var pyItem = interpreter.toPython(item);
+      var g = PyGILState_Ensure();
+      defer PyGILState_Release(g);
+
+      var pyItem = interpreter.toPythonInner(item);
       PySequence_SetItem(this.getPyObject(), idx.safeCast(Py_ssize_t), pyItem);
       this.check();
     }
   }
 
 
+  // TODO: now that we handle this on every call to the python interpreter,
+  // is this still needed as a user facing type?
```

Comment:
I would argue no.  But I've of two minds whether I think it should be handled in this PR:
- Pros:
  - All the related changes are in one place, which makes it easier to see the link between actions.  If we do another full overhaul, we'll know to look at this type again if we start by looking at this PR
- Cons:
  - If we do end up backing away from this change, we might not remember to revisit whether this type should be public again, so users might not have documentation for any strategy.

Reply:
Seeing that fromPickle still needs the GIL makes me lean more towards "don't hide it yet".  But that probably means we should update the documentation for it to point back to the earlier docs about the more recommended strategy

Reply:
`fromPickle` only needs it because its an internal function that is meant to be called with the GIL already held. Those tests are testing undocumented/private API details.

But I take your point that we should not hide it yet, which is also my inclination. 

I'll note that `GIL` can be used freely by users if they so choose, `threadState` is likely to crash their program